pyccel/codegen/printing/cwrappercode.py

```diff
@@ -874,6 +882,12 @@ def _print_FunctionDef(self, expr):
 
         # Call free function for python type
         wrapper_body += [FunctionCall(Py_DECREF, [i]) for i in self._to_free_PyObject_list]
+        # Call free function for C type
+        if self._target_language == 'c':
+            wrapper_body += [FunctionCall(C_Free, [DottedVariable(i,
```

Comment:
Where are these objects allocated?

Reply:
in the ndarrays library, we free the ndarrays strides after finishing work with the casted numpy array.
check this function `numpy_to_ndarray_strides` located in `stdlib/ndarrays/ndarrays.c`

Reply:
If you allocate the `shape` you also need to free this. Wouldn't using the `Deallocate` node and marking this variable as a pointer get the correct deallocation?